lib/ansible/modules/cloud/amazon/ec2_vpc_route_table_info.py

```diff
@@ -54,80 +62,187 @@
+      vpc_id:
+        description: The ID of the VPC peering connection.
+        returned: when vpc peering available
+        type: str
+        sample: vpc-a01106c2
+'''
 
+import json
 
-def get_route_table_info(route_table):
+try:
+    from botocore.exceptions import ClientError, BotoCoreError
+except ImportError:
+    pass  # caught by AnsibleAWSModule
 
-    # Add any routes to array
-    routes = []
-    associations = []
-    for route in route_table.routes:
-        routes.append(route.__dict__)
-    for association in route_table.associations:
-        associations.append(association.__dict__)
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (ansible_dict_to_boto3_filter_list, camel_dict_to_snake_dict,
+                                      boto3_tag_list_to_ansible_dict)
 
-    route_table_info = {'id': route_table.id,
-                        'routes': routes,
-                        'associations': associations,
-                        'tags': route_table.tags,
-                        'vpc_id': route_table.vpc_id
-                        }
 
-    return route_table_info
+def date_handler(obj):
+    return obj.isoformat() if hasattr(obj, 'isoformat') else obj
 
 
-def list_ec2_vpc_route_tables(connection, module):
+def list_route_tables(route_table, module):
+    params = dict()
 
-    filters = module.params.get("filters")
-    route_table_dict_array = []
+    params['Filters'] = ansible_dict_to_boto3_filter_list(module.params.get('filters'))
+    params['RouteTableIds'] = module.params.get('route_table_ids')
 
     try:
-        all_route_tables = connection.get_all_route_tables(filters=filters)
-    except BotoServerError as e:
-        module.fail_json(msg=e.message)
-
-    for route_table in all_route_tables:
-        route_table_dict_array.append(get_route_table_info(route_table))
-
-    module.exit_json(route_tables=route_table_dict_array)
+        result = json.loads(json.dumps(route_table.describe_route_tables(**params), default=date_handler))
+    except ValueError as e:
+        module.fail_json_aws(e, msg="Cannot validate JSON data")
+    except (ClientError, BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Could not describe route tables")
+    snaked_route_tables = [camel_dict_to_snake_dict(route_table,
+                           ignore_list=['Tags']) for route_table in result['RouteTables']]
+    if snaked_route_tables:
+        for route_table in snaked_route_tables:
+            route_table['id'] = route_table['route_table_id']
+            for association in route_table['associations']:
+                association['id'] = association['route_table_association_id']
+                if 'subnet_id' not in association:
+                    association['subnet_id'] = None
+            for route in route_table['routes']:
+                for key in ['instance_id']:
+                    if key not in route:
+                        route[key] = None
```

Comment:
Hi @tremble, this is the part where we add some keys to the returned dict to make it backwards compatible. Boto always had these keys defined and they are set to None when necessary. Now in boto3, these keys are missing from the dict if they are not applicable. The 2 tests that I added ensure that we maintain backwards compatibility by always having those keys present in the returned dict. When these keys are added their values are set to None to mimic boto2 behavior. So what tests exactly do I need to add here?